### PR TITLE
Add day 18 explanations

### PR DESCRIPTION
**Pull request by EugeneFlesselle**

### CODE REVIEW

File: docs/2023/puzzles/day18.md [hunk 1]
```diff
@@ -2,10 +2,136 @@ import Solver from "../../../../../website/src/components/Solver.js"
...
+
+Provided our current position, we can then keep track of the lagoon area as follows:
+- When going `Right`: we count all the points in the line we cover, i.e the `length` of the trench.
+- When going `Down`: we count all the points which we leave on the left (or pass over),
+  i.e. the `length` of the downwards trench times our current `x` coordinate,
+  `+1` to count the current vertical trench as in the area.
+  Of course, we may be including too much at this point,
+  since we do not yet know what part of the left is actually in the lagoon,
+  but we will account for it later.
+- When going `Right`: there is nothing to add,
```
Comment (bishabosha) [inline docs/2023/puzzles/day18.md:33]: is this meant to be `Left`?

Comment (bishabosha) [inline docs/2023/puzzles/day18.md:33]:
```suggestion
- When going `Left`: there is nothing to add,
```
I can see from the code that it is

**Review by bishabosha — APPROVED** [review]
Thanks, looks like a simpler one

File: docs/2023/puzzles/day18.md (hunk 1 above)
Comment (EugeneFlesselle) [reply docs/2023/puzzles/day18.md]: Yes definitely, good catch

**Pull request merged**